exporters/otlp/include/opentelemetry/exporters/otlp/otlp_http_client.h

```diff
@@ -99,6 +109,8 @@ class OtlpHttpClient
    */
   explicit OtlpHttpClient(OtlpHttpClientOptions &&options);
 
+  ~OtlpHttpClient();
```

Comment:
GC here mean garbage collection of http sessions.When a http session is finished, we can not destroy it immediately right now, because it may still be visited in some codes. (e.g. When we remove a session in OnResponse, it will be visited in OnEvent later).
So I move the finished sessions into `gc_sessions_` and real destroy it soon later.